### PR TITLE
Implement new window layouter

### LINKED ISSUE #3: client window positioning not working with <3 or =4 clients
For <3 clients (ssh targets) the first target is almost entirely maximized while other clients and the daemon are outside of the visible space.
For =4 clients, the clients fill the available desktop space and the daemon is rendered outside of the visible space.

### PR DESCRIPTION
**Pull request by timoschwarzer**
Closes #3

### CODE REVIEW

**Comment by timoschwarzer** [comment]
Closes #3

File: src/daemon/main.rs [hunk 1]
```diff
@@ -157,18 +158,41 @@ impl Daemon {
...
-    // FIXME: if number of hosts is < 3 or =5 determination of client spacial attributes falls apart
-    let height_width_ratio = workspace_area.height as f64 / workspace_area.width as f64;
-    let number_of_columns = (number_of_consoles as f64 / height_width_ratio).sqrt() as i32;
-    let console_width = workspace_area.width / number_of_columns;
-    let console_height = (console_width as f64 * height_width_ratio) as i32;
-    let x = workspace_area.width / number_of_columns * (index % number_of_columns);
-    let y = index / number_of_columns * console_height;
+    let aspect_ratio = workspace_area.width as f64 / workspace_area.height as f64;
+
+    const ASPECT_RATIO_ADJUSTMENT: f64 = 0.5;
```
Comment (whme) [inline src/daemon/main.rs:168]: How about moving this to the top of the file below the KEY_EVENT constant?
I would also suggest `0f64` as default value, what do you think?

**Review by whme — CHANGES_REQUESTED** [review]
The daemon console is now always rendered outside the visible desktop space.
Once that's fixed I think we can merge it.

File: src/daemon/main.rs (hunk 1 above)
Comment (timoschwarzer) [reply src/daemon/main.rs]: Done

**Comment by timoschwarzer** [comment]
> The daemon console is now always rendered outside the visible desktop space. Once that's fixed I think we can merge it.

Fixed.

**Comment by timoschwarzer** [comment]
Superseded by #8

**Pull request closed without merging**